blog-posts/sls-security-headers/sls-security-headers.md

```diff
@@ -0,0 +1,91 @@
+---
+published: true
+title: 'Deliver perfect HTTP security headers with AWS CloudFront'
+cover_image: 'https://dev-to-uploads.s3.amazonaws.com/uploads/articles/q5f8qmpiczu9jzvx6mb2.png'
+description: 'HTTP security headers protects your users from common attacks. AWS CloudFront makes it easy to add HTTP security headers to your application.'
+tags: cloudfront, security, AWS, audit
+canonical_url:
+---
+
+– Did you just say security headers?
+
+– Yes.
+
+– Are you talking about CORS headers?
+
+– No.
+
+## TL;DR
+
+AWS CloudFront enables you to easily make your users browse safer with managed security headers. Quickly check your compliance with [sls-mentor](https://www.sls-mentor.dev/).
+
+## HTTP security headers: a very simple addition to your website to efficiently prevent attacks
+
+In the last 5 years, have worked in a dozen teams to develop and run web applications. However, I have learnt about **HTTP security headers** very late in my full stack developer journey. This is only very recently, when I joined [Theodo](https://www.theodo.fr/), that I was explained what HTTP security headers actually are. 😮
+
+HTTP security headers are a mechanism to **protect your application from common attacks**. They are, as their name suggests, a set of HTTP headers with security purposes. They are set by the server and read by the browser, which then applies the **security policy** described by the header.
+
+HTTP security headers are good *deployment* practices. Thus, they fall under the responsibility of ops teams, rather than dev teams. But more and more, dev teams are also responsible for the deployment of their applications.
+
+The table below shows a subset of HTTP security headers. The description mentions the common attacks that the header protects against. The recommended value column contains optimal header values to protect your users against these threats.
+
+| Name | Value | Description |
+| --- | --- | --- |
+| [Strict-Transport-Security](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security) | `max-age=31536000` | Informs browsers that the site should only be accessed using HTTPS, and that any future attempts to access it using HTTP should automatically be converted to HTTPS. The user's inputs will never be readable by a man on the middle (except potentially the first call ever, which does not contain any sensitive information). |
+| [X-Content-Type-Options](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Content-Type-Options) | `nosniff` | Prevents older browsers from sniffing the content of files sent by the server rather than trusting the Content-Type header. This prevents the detection and execution of malicious Javascript in an image uploaded by an attacker and then retrieved by a victim. |
```

Comment:
`Prevents older browsers from sniffing the content of files sent by the server rather than trusting the Content-Type header. This prevents the detection and execution of malicious Javascript in an image uploaded by an attacker and then retrieved by a victim.` => Est ce que sniffing c'est assez clair? Propositions `guessing`, `speculate`